src/mds/Server.cc

```diff
@@ -3899,6 +3899,11 @@ void Server::handle_client_lookup_ino(MDRequestRef& mdr,
     return _lookup_snap_ino(mdr);
 
   inodeno_t ino = req->get_filepath().get_ino();
+  auto _ino = ino.val;
+  if (_ino < MDS_INO_SYSTEM_BASE && _ino != MDS_INO_ROOT) {
+    respond_to_request(mdr, -CEPHFS_ESTALE);
+    return;
+  }
```

Comment:
nit: a comment about why we do this wouldn't be a bad idea here, but looks good otherwise.

Reply:
done